### PR TITLE
Initial cull of master term in relation to git

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

Replaces key occurrences of master with main in dev-docs and buildscript. Note that this is not everything in the codebase just important buildscripts and dev-docs so that developers see up-to-date information.

Other occurrences will be dealt with separately.

**To test:**

* Review 

*There is no associated issue.*

*This does not require release notes* because **it is an internal change.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by peterfpeterson — DISMISSED** [review]
I appreciate the couple of formatting additions hiding in here

File: dev-docs/source/BuildingWithCMake.rst [hunk 1]
```diff
@@ -8,7 +8,7 @@ Building with CMake
   :local:
 
 CMake is the build system for the entirety of Mantid (Framework, MantidQt and MantidWorkbench). It is used to generate native build files for your platform, which can be Makefiles (for use with make, nmake or jom) for command line builds or project/solution files for an IDE (e.g. Visual Studio, Eclipse, Qt Creator, XCode).
-For a "how is it used version" of this guide, look at the scripts used on the builservers for `windows <https://github.com/mantidproject/mantid/blob/master/buildconfig/Jenkins/buildscript.bat>`_ or `linux/osx <https://github.com/mantidproject/mantid/blob/master/buildconfig/Jenkins/buildscript>`_.
+For a "how is it used version" of this guide, look at the scripts used on the builservers for `windows <https://github.com/mantidproject/mantid/blob/main//buildconfig/Jenkins/buildscript.bat>`_ or `linux/osx <https://github.com/mantidproject/mantid/blob/master/buildconfig/Jenkins/buildscript>`_.
```
Comment (DavidFair) [inline dev-docs/source/BuildingWithCMake.rst:11]: Looks like this one has slipped through

File: dev-docs/source/DataFilesForTesting.rst [hunk 2]
```diff
@@ -101,9 +101,9 @@ Using Existing Data
 
 There are two places files may be found:
 
-- `.../Testing/Data/ <https://github.com/mantidproject/mantid/tree/master/Testing/Data>`__
+- `.../Testing/Data/ <https://github.com/mantidproject/mantid/tree/main//Testing/Data>`__
```
Comment (DavidFair) [inline dev-docs/source/DataFilesForTesting.rst:104]: Some double `//` have slipped in here

File: dev-docs/source/WritingPerformanceTests.rst [hunk 3]
```diff
@@ -141,11 +141,11 @@ There is a job in Jenkins (our continuous integration system) that runs
 the performance test suite and generates output that enables us to
 easily monitor timings. The job runs a set of `performance tests on the
 master branch of
-Mantid <http://builds.mantidproject.org/job/master_performancetests2/>`__.
+Mantid <http://builds.mantidproject.org/job/main/_performancetests2/>`__.
```
Comment (DavidFair) [inline dev-docs/source/WritingPerformanceTests.rst:144]: This has an extra slash in the job name now

**Review by DavidFair — CHANGES_REQUESTED** [review]
Some minor replacement mistakes, otherwise looks good

**Review by DavidFair — APPROVED** [review]
LGTM thanks for doing those, lets :shipit:

**Pull request merged**